### PR TITLE
SB 245,247,248 Nurse alert notification

### PR DESCRIPTION
**Pull request by Yqiyuan**
# Description
SB 245 The alert disappears after two minutes
SB 247 Keep Track of Alert Acknowledgments
SB 248 Add Alert Priority Rule

https://user-images.githubusercontent.com/12007072/228959354-e6bce85c-5935-4541-ace4-cb3442c7b0ce.mp4

# Test Plan

Please explain how this change will be tested.

- [x] Unit test
- [x] Integration test (including unit tests with supertest)
- [x] Manual test (Please explain below for the test steps)
- [ ] Not Applicable

# Task Link

Please add related jira task links

### CODE REVIEW

File: client/src/utils/DataStore.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export interface IGroupNotification {
   notificationType: string
   channelId: string
   messageId: string
+  detailedContent?: any
```
Comment (bettsongy) [inline client/src/utils/DataStore.ts:17]: It would be better to define a more specific type for detailedContent if possible.

Comment (Yqiyuan) [reply client/src/utils/DataStore.ts:17]: The content of detailedContent is not fixed yet. So I use any for now

**Review by bettsongy — APPROVED** [review]

File: client/src/utils/Notification.tsx [hunk 2]
```diff
@@ -62,26 +69,60 @@ export default function Notification() {
...
-            clearGroupNotification()
-          }}
-        >
-          <FlashScreen alertType={notification?.alert.type} />
-        </Backdrop>
-      </Link>
+      <Backdrop
+        sx={{ color: '#fff', zIndex: (theme) => theme.zIndex.tooltip * 2 }}
+        open={!!notification}
+        onClick={(event) => {
```
Comment (jennyzeng) [inline client/src/utils/Notification.tsx:75]: consider abstract this one out into a new function vs nested here

**Review by jennyzeng — APPROVED** [review]

File: client/src/utils/Notification.tsx (hunk 2 above)
Comment (Yqiyuan) [reply client/src/utils/Notification.tsx]: Good suggestion. Method extracted.

File: client/src/components/Alerts/FlashScreen.tsx [hunk 3]
```diff
@@ -1,25 +1,62 @@
 import React, { FunctionComponent } from 'react'
-import { Card, Typography } from '@mui/material'
+import { Button, Card, Stack, Typography } from '@mui/material'
 import useAlertTheme from '../../hooks/useAlertTheme'
 import * as _ from 'lodash'
+import { ALERTS } from '../../utils/Alerts'
 
 interface FlashScreenProps {
   alertType?: string | undefined
+  detailedContent?: any
```
[the remark this answers is not in the document]
Comment (Yqiyuan) [reply client/src/components/Alerts/FlashScreen.tsx]: Added current interface. May need to modify it later

File: client/src/utils/Notification.tsx [hunk 4]
```diff
@@ -36,22 +37,28 @@ export default function Notification() {
...
           channelId: groupNotification.channelId,
           messageId: groupNotification.messageId,
+          detailedContent: groupNotification.detailedContent,
         })
     }
     if (messageNotification) {
       setMsgNotification(messageNotification)
     }
+    if (notification && groupNotification === null) {
+      setNotification(null)
```
[the remark this answers is not in the document]
Comment (Yqiyuan) [reply client/src/utils/Notification.tsx:47]: This has dependencies with other functionalities. I will treat it as a technical debt

**Pull request merged**